google-cloud-bigtable/lib/google-cloud-bigtable.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# This file is here to be autorequired by bundler, so that the
+# Google::Cloud#bigtable method can be available,
+# but the library and all dependencies won't be loaded until required and used.
+
+require "google/cloud/bigtable"
+require "bigtable"
+
+module Google
+  module Cloud
```

Comment:
So the `Google::Cloud#bigtable` creation method will added later?

Reply:
Yes, I am going to add `Google::Cloud#bigtable` method for creating a client. It is in a different branch for data operations https://github.com/jiren/google-cloud-ruby/blob/bigtable-data-api-wrapper/google-cloud-bigtable/lib/google-cloud-bigtable.rb 

I will merge it to the admin api branch one all data operation functionalities are completed.

Reply:
Will `Google::Cloud#bigtable` return an instance client, table client, or a data client?

Reply:
Will this gem also be adding a dependency on `google-cloud-core`?

Reply:
@blowmage  Yes, `Google::Cloud#bigtable` will return client based on `client_type` argument. Types are `: data`, `:table`, `:instance`. Default client type is `:data`.

I am planning to to add `google-cloud-core`. It looks better solution to create service client object and for error classes.